Compilation failure on newer versions of HDF5

@mhmerrill was getting a compilation failure with HDF5 version 1.12.1 and it looks like the macro compatibility of HDF5 between versions 1.10.x and 1.12.x  has an issue with function H5Oget_info_by_name (versions 1 vs 3, the proxy function now points to 3 which takes more args).

For now we should explicitly call version 1.

This doesn't make me feel good :(

```
${HOME}/projects/arkouda//src/c_helpers/help_h5ls.c:16:49: warning: incompatible pointer types passing 'H5O_info2_t *' (aka 'struct H5O_info2_t *') to parameter of type 'H5O_info1_t *' (aka 'struct H5O_info1_t *') [-Wincompatible-pointer-types]
    status = H5Oget_info_by_name1(loc_id, name, info_t_ptr, H5P_DEFAULT);
                                                ^~~~~~~~~~
${HOME}/opt/hdf5-1.12.1/include/H5Opublic.h:2052:81: note: passing argument to parameter 'oinfo' here
H5_DLL herr_t H5Oget_info_by_name1(hid_t loc_id, const char *name, H5O_info1_t *oinfo, hid_t lapl_id);
```

On a build of HDF5 v1.12.1 (using default options) it looks like the proxied type-struct for H5O_info_t was updated to version 2 ... but version 1 doesn't actually exist by name in 1.10.6 so we can't explicitly point to it by name.  I'm not sure if this is the issue that Mike ran into, but it's in the general area of the failure he experienced.

--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,10 @@ default: $(DEFAULT_TARGET)
 VERBOSE ?= 0
 
 CHPL := chpl
+
+# We need to make the HDF5 API use the 1.10.x version for compatibility between 1.10 and 1.12
+CHPL_FLAGS += --ccflags="-DH5_USE_110_API"
+
 CHPL_DEBUG_FLAGS += --print-passes
 ifdef ARKOUDA_DEVELOPER
 CHPL_FLAGS += --ccflags="-O1"
@@ -248,6 +252,7 @@ else
 	ARKOUDA_COMPAT_MODULES += -M $(ARKOUDA_SOURCE_DIR)/compat/lt-125
 endif
 
+# This is the main compilation statement section
 $(ARKOUDA_MAIN_MODULE): check-deps $(ARROW_O) $(ARKOUDA_SOURCES) $(ARKOUDA_MAKEFILES)
 	$(CHPL) $(CHPL_DEBUG_FLAGS) $(PRINT_PASSES_FLAGS) $(REGEX_MAX_CAPTURES_FLAG) $(OPTIONAL_SERVER_FLAGS) $(CHPL_FLAGS_WITH_VERSION) $(ARKOUDA_MAIN_SOURCE) $(ARKOUDA_COMPAT_MODULES) -o $@
 

--- a/src/c_helpers/help_h5ls.c
+++ b/src/c_helpers/help_h5ls.c
@@ -12,7 +12,8 @@ herr_t c_get_HDF5_obj_type(hid_t loc_id, const char *name, H5O_type_t *obj_type)
 {
     herr_t status;
     H5O_info_t info_t;
-    status = H5Oget_info_by_name1(loc_id, name, &info_t, H5P_DEFAULT);
+    H5O_info_t* info_t_ptr = &info_t;
+    status = H5Oget_info_by_name(loc_id, name, info_t_ptr, H5P_DEFAULT);
     *obj_type = info_t.type;
     return status;
 }

